src/main/kotlin/skytils/skytilsmod/listeners/ChatListener.kt

```diff
@@ -211,6 +252,262 @@ class ChatListener {
+                SkillUtils.calcXpWithProgress(catacombsObj.experience ?: 0.0, SkillUtils.dungeoneeringXp.values)
+
+            val secrets = playerResponse.achievements.getOrDefault("skyblock_treasure_hunter", 0)
+
+            val component = UMessage("&2&l-----------------------------\n")
+                .append(
+                    UTextComponent(playerResponse.rankPrefix
+                            + if(playerResponse.rank.toString() == "NONE") {""} else {" "}
+                            + "${playerResponse.player.getString("displayname")}'s §dStats: §7(Hover)\n").setHoverText("§7Click to run: /skytilscata $username").setClick(
+                        ClickEvent.Action.RUN_COMMAND, "/skytilscata $username")
+                )
+                .append("§dCata: §b${NumberUtil.nf.format(floor(cataLevel))}\n\n")
+            val armor = profileData.armor
+            val armorArray:ArrayList<List<String>> = ArrayList()
+            armor?.forEveryItem {
+                val armorLore = it.asMinecraft.getTooltip(mc.thePlayer,false)
+                armorArray.add(armorLore)
+            }
+            armorArray.reverse()
+            for(armorPiece in armorArray) {
+                val armorName = armorPiece[0]
+                component.append(UTextComponent(armorName + "\n").setHoverText(buildString {
+                    for (line in armorPiece) {
+                        append(line + "\n")
+                    }
+                }))
+            }
+            val pets = profileData.pets
+            var activePet: Pet? = null
+            for (pet in pets) {
+                if (pet.active) {
+                    activePet = pet
+                }
+            }
+            if (activePet != null) {
+                var rarity = "§f"
+                if (activePet.tier.toString().uppercase() == "UNCOMMON") {
+                    rarity = "§a"
+                } else if (activePet.tier.toString().uppercase() == "RARE") {
+                    rarity = "§9"
+                } else if (activePet.tier.toString().uppercase() == "EPIC") {
+                    rarity = "§5"
+                } else if (activePet.tier.toString().uppercase() == "LEGENDARY") {
+                    rarity = "§6"
+                } else if (activePet.tier.toString().uppercase() == "MYTHIC") {
+                    rarity = "§d"
+                }
+
+                component.append(UTextComponent(rarity + activePet.type.toString().lowercase().replace("_", " ")
+                    .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
+                        + "\n\n")
+                    .setHoverText( if(activePet.heldItem == null) {"§cNo Pet Item"} else "§b" +
+                            activePet.heldItem.toString().lowercase().replace("pet_item_", "").replace("_", " ")
+                                .replaceFirstChar { if (it.isLowerCase()) it.titlecase(Locale.getDefault()) else it.toString()}
```

Comment:
Why is there a locale supplied here

Reply:
Ide was crying about me just doing it normally and suggested me to do this